tests/datavolume_test.go

```diff
@@ -1013,15 +1013,23 @@ var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]DataVolume tests",
 			Expect(err).ToNot(HaveOccurred())
 			f.ForceBindIfWaitForFirstConsumer(pvc)
 
-			By("find importer pod")
-			pod, err := utils.FindPodByPrefix(f.K8sClient, dataVolume.Namespace, common.ImporterPodName, common.CDILabelSelector)
+			By("verifying the Datavolume is not complete yet")
+			foundDv, err := f.CdiClient.CdiV1beta1().DataVolumes(dataVolume.Namespace).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
-			Expect(pod).ToNot(BeNil())
-			By("verifying passed annotation")
-			Expect(pod.Annotations[controller.AnnPodNetwork]).To(Equal("net1"))
-			By("verifying non-passed annotation")
-			Expect(pod.Annotations["annot1"]).ToNot(Equal("value1"))
+			if foundDv.Status.Phase != cdiv1.Succeeded {
+				By("find importer pod")
+				var sourcePod *v1.Pod
+				Eventually(func() bool {
+					sourcePod, err = utils.FindPodByPrefix(f.K8sClient, dataVolume.Namespace, common.ImporterPodName, common.CDILabelSelector)
+					return err == nil
+				}, timeout, pollingInterval).Should(BeTrue())
+				By("verifying passed annotation")
+				Expect(sourcePod.Annotations[controller.AnnPodNetwork]).To(Equal("net1"))
+				By("verifying non-passed annotation")
+				Expect(sourcePod.Annotations["annot1"]).ToNot(Equal("value1"))
```

Comment:
on one hand the 8 lines here it could be a helper function to check for annotations on a pod (and reused in two other places), on the other hand I like how tests are verbose and expectations are in place right now .

Reply:
I can make it a function scoped to the describe that uses it. should still be readable.